ingest-app/src/cmr/ingest/services/ingest_service.clj

```diff
@@ -391,3 +396,119 @@
                            msg/token-required-for-variable-modification))
           (dissoc :revision-date :transaction-id)
           (update-in [:revision-id] inc)))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Service Ingest Functions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(def ^:private update-service-validations
+  "Service level validations when updating a service."
+  [(cv/field-cannot-be-changed :service-name)
+   ;; Originator id cannot change but we allow it if they don't specify a
+   ;; value.
+   (cv/field-cannot-be-changed :originator-id true)])
+
+(defn validate-update-service
+  "Validates a service update."
+  [existing-service updated-service]
+  (cv/validate!
+   update-service-validations
+   (assoc updated-service :existing existing-service)))
+
+(defn overwrite-service-tombstone
+  "This function is called when the service exists but was previously
+  deleted."
+  [context concept service user-id]
+  (mdb2/save-concept context
+                    (-> concept
+                        (assoc :metadata (pr-str service)
+                               :deleted false
+                               :user-id user-id)
+                        (dissoc :revision-date)
+                        (update-in [:revision-id] inc))))
+
+(defn- service->new-concept
+  "Converts a service into a new concept that can be persisted in metadata
+  db."
+  [service]
+  {:concept-type :service
+   :native-id (:name service)
+   :metadata (pr-str service)
+   :user-id (:originator-id service)
+   ;; The first version of a service should always be revision id 1. We
+   ;; always specify a revision id when saving variables to help avoid
+   ;; conflicts
+   :revision-id 1
+   :format mt/edn
+   :provider-id "CMR"              ; XXX pending changes from @ygliuvt
+   :extra-fields {
+     :entry-id (:name service)     ; XXX pending changes from @ygliuvt
```

Comment:
Update: we chatted about this, and these issues should be resolved after @ygliuvt's metadata-db PR for services is merged.